### PR TITLE
tools/tcpretrans: add tracepoint support

### PR DESCRIPTION
**Pull request by boat0**
If tracepoint tcp:tcp_retransmit_skb exists (kernel version >=4.15)
we use tracepoint instead of kprobe for efficiency.

Co-authored-by: Runlong Lin <linrunlong@bytedance.com>

### CODE REVIEW

**Comment by yonghong-song** [comment]
[buildbot, test this please]

File: tools/tcpretrans.py [hunk 1]
```diff
@@ -187,11 +252,19 @@
     bpf_text = bpf_text.replace("IPV6_INIT", struct_init['ipv6']['count'])
     bpf_text = bpf_text.replace("IPV4_CORE", count_core_base.replace("COUNT_STRUCT", 'ipv4_count'))
     bpf_text = bpf_text.replace("IPV6_CORE", count_core_base.replace("COUNT_STRUCT", 'ipv6_count'))
+    if BPF.tracepoint_exists("tcp", "tcp_retransmit_skb"):
+        bpf_text_tracepoint = bpf_text_tracepoint.replace("IPV4_CODE", struct_init_tracepoint['ipv4']['count'])
+        bpf_text_tracepoint = bpf_text_tracepoint.replace("IPV6_CODE", struct_init_tracepoint['ipv6']['count'])
+        bpf_text += bpf_text_tracepoint
```
Comment (yonghong-song) [inline tools/tcpretrans.py]: Looks good in general. Could you do a little bit refactoring not to load `trace_retransmit` function if tracepoint exists? If would be confusing if both `trace_retransmit` and the tracepoint program are loaded.

Comment (boat0) [reply tools/tcpretrans.py]: Okay. Either kprobe or tracepoint will be loaded, but not both, given '-l' is not there.

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Comment by yonghong-song** [comment]
The change looks good. Maybe we can just call `BPF.tracepoint_exists("tcp", "tcp_retransmit_skb")` once and remember the answer instead of calling multiple times. But this is minor.

**Pull request merged**